framework/sandstone_tests.cpp

```diff
@@ -71,34 +71,54 @@ std::vector<struct test *> SandstoneTestSet::lookup(const char *name)
     return res;
 }
 
-SandstoneTestSet::SandstoneTestSet(struct test_set_cfg cfg, unsigned int flags) : cfg(cfg), flags(flags) {
-    load_all_tests(); /* initialize the catalog */
-    if (!(flags & enable_all_tests)) return;
-    std::span<struct test> source = !cfg.is_selftest ? regular_tests : selftests;
-    for (struct test &test : source) {
-        struct test_cfg_info ti;
-        ti.status = test_cfg_info::enabled;
-        ti.test = &test;
-        test_set.push_back(&test);
-    }
-};
-
-struct test_cfg_info SandstoneTestSet::add(test_cfg_info t)
+int SandstoneTestSet::add(test_cfg_info t)
 {
     struct test_cfg_info &ti = test_set.emplace_back(t);
+
+    auto it = std::remove_if(test_set.begin(), test_set.end(), [&](const test_cfg_info &ti) {
+        return ti.test == t.test;
+    });
```

Comment:
This moves the matching tests to the end. Why are we doing that? Wouldn't this make
```
  -e A -e B -e A
```
execute "B, A, A" ?

Did you mean to insert this loop inside the `if` below? Maybe move the `remove_if` + `erase` operations into a lambda.

And why are we `add`ing disabled testconfigs?

Reply:
I removed confusing part of the code. I meant not to add tests that are already disabled. 

I took the lambda from `remove` method. The idea here is to remove for the `test_set` all tests that match `disable_by_feature` or not match `enable_by_feature`